starlite/middleware/exceptions/debug_response.py

```diff
@@ -0,0 +1,123 @@
+from html import escape
+from inspect import getinnerframes
+from pathlib import Path
+from traceback import format_exception
+from typing import TYPE_CHECKING, List
+
+from starlite.enums import MediaType
+from starlite.response import Response
+from starlite.status_codes import HTTP_500_INTERNAL_SERVER_ERROR
+
+if TYPE_CHECKING:
+    from inspect import FrameInfo
+
+    from starlite.connection import Request
+
+tpl_dir = Path(__file__).parent / "templates"
+
+
+def get_symbol_name(frame: "FrameInfo") -> str:
+    """Return full name of the function that is being executed by the given
+    frame."""
+    classname = ""
+    locals_dict = frame.frame.f_locals
+    # probably we are inside of an instance method
+    # this piece assumes that the code uses standard names "self" and "cls"
+    # in instance and class methods
+    if locals_dict.get("self"):
+        classname = f"{type(locals_dict['self']).__name__}."
+    # inside a class method?
+    elif locals_dict.get("cls"):
+        classname = f"{locals_dict['cls'].__name__}."
+
+    return f"{classname}{frame.function}"
+
+
+def create_line_html(
+    line: str,
+    line_no: int,
+    frame_index: int,
+    idx: int,
+) -> str:
+    """Return html representation of a line including real line number in the
+    source code."""
+    template = '<tr class="{line_class}"><td class="line_no">{line_no}</td><td class="code_line">{line}</td></tr>'
+    data = {
+        "line_no": line_no - frame_index + idx,
+        "line": escape(line).replace(" ", "&nbsp"),
+        "line_class": "current_line" if idx == frame_index else "",
+    }
+    return template.format(**data)
+
+
+def create_frame_html(frame: "FrameInfo", collapsed: bool) -> str:
+    """Return html representation of the given frame object including filename
+    containing source code and name of the function being executed."""
+    frame_tpl = (tpl_dir / "frame.html").read_text()
+
+    code_lines: List[str] = []
+    for idx, line in enumerate(frame.code_context or []):
+        code_lines.append(create_line_html(line, frame.lineno, frame.index or 0, idx))
+
+    data = {
+        "file": escape(frame.filename),
+        "line": frame.lineno,
+        "symbol_name": escape(get_symbol_name(frame)),
+        "code": "".join(code_lines),
+        "frame_class": "collapsed" if collapsed else "",
+    }
+    return frame_tpl.format(**data)
+
+
+def create_exception_html(exc: BaseException, frame_limit: int) -> str:
+    """Return html representation of exception frames."""
+    frames = getinnerframes(exc.__traceback__, frame_limit) if exc.__traceback__ else []
+    result = []
+    for idx, frame in enumerate(reversed(frames)):
+        result.append(create_frame_html(frame=frame, collapsed=(idx > 0)))
+
+    return "".join(result)
+
+
+def create_html_response_content(exc: Exception, request: "Request", frame_limit: int = 15) -> str:
```

Comment:
i wonder whether we should also handle `ExceptionGroups`, which would require we use a backport for this.

Reply:
I was thinking about this but decided not to handle groups in the Starlite since the framework  itself and its dependencies, nor other major libs,  won't be throwing `ExceptionGroup`s in the   near future.

The only source of them would be the user code and I don't think it is good idea to add another package to the dependencies list only to show more details about a a group.

I am working on a rich exception renderer which will be a separate package. I suggest to handle exception groups in this package and direct users willing to see nice output for their exception groups to install the package.